include/eve/arch/riscv/tags.hpp

```diff
@@ -0,0 +1,96 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/cardinals.hpp>
+#include <eve/arch/cpu/tags.hpp>
+#include <eve/arch/riscv/predef.hpp>
+#include <eve/detail/meta.hpp>
+
+#include <optional>
+
+namespace eve
+{
+//================================================================================================
+// ABI tags for all RISCV bits SIMD registers
+//================================================================================================
+template<std::size_t VectorRegisterBitSize> struct rvv_abi_
+{
+  static constexpr std::size_t MaxLmul         = 8;
+  static constexpr std::size_t bits            = VectorRegisterBitSize * MaxLmul;
+  static constexpr std::size_t bytes           = bits / 8;
+  static constexpr bool        is_wide_logical = false;
+
+  template<typename Type>
+  static constexpr std::size_t fundamental_vector_size = VectorRegisterBitSize * sizeof(Type) / 8;
+
+  template<typename Type>
+  static constexpr std::size_t fundamental_cardinal =
+      fundamental_vector_size<Type> / sizeof(Type) / 8;
+  // Work in terms of vector intrinsics types
+  template<typename Type>
+  static constexpr bool is_full = ((Type::size() * sizeof(typename Type::value_type))
+                                   >= fundamental_vector_size<typename Type::value_type> / 8);
+
+  template<typename Type>
+  static constexpr std::size_t expected_cardinal =
+      VectorRegisterBitSize / 8 / sizeof(Type) * MaxLmul;
+
+  // return natural lmul if > 0, frac otherwise
+  template<typename Type> static consteval int getLMUL(std::ptrdiff_t cardinal)
+  {
+    constexpr std::ptrdiff_t m1_len       = VectorRegisterBitSize;
+    constexpr std::ptrdiff_t min_len      = m1_len * sizeof(Type) / 8;
+    std::ptrdiff_t           expected_len = sizeof(Type) * 8 * cardinal;
+    std::ptrdiff_t           reg_len      = std::max(min_len, expected_len);
+    if( reg_len == m1_len ) return 1;
+    else if( reg_len > m1_len ) return static_cast<int>(reg_len / m1_len);
+    else return -static_cast<int>(m1_len / reg_len);
+  }
+
+  template<typename Type> static consteval std::ptrdiff_t getLogicalRatio(std::ptrdiff_t cardinal)
+  {
+    auto           lmul         = getLMUL<Type>(cardinal);
+    constexpr auto element_size = sizeof(Type) * 8;
+    return lmul > 0 ? element_size / lmul : element_size * (-lmul);
+  }
```

Comment:
eve uses `_` notations.

This is what I suggest.

Similar to https://github.com/jfalcou/eve/blob/main/include/eve/arch/arm/sve/sve_true.hpp

Create `eve/arch/riscv/ratios.hpp` or smth

in that file:

```
template <simd_value T>
constexpr rvv_lmul_v = []{
  // ...
}();

template <logical_simd_value T>
constexpr rvv_logical_ratio_v = []{
  // ...
}();
```

most places - use those